_plugins/server_generator.rb

```diff
@@ -1,5 +1,7 @@
 module Jekyll
   class ServerPage < Page
+    include Jekyll::ThanksToFilter
```

Comment:
Ah I missed that... I guess that also why there's that weird code to find the site details in the filter, because it isn't always a filter?

No idea where library code would go really - maybe @gravitystorm knows?